doc/new-guidelines.md

```diff
@@ -0,0 +1,467 @@
+ (fn [{:keys [db] :as cofx}]
+   ...))
+
+;; good
+(fx/defn save-all
+  {:events [:wakuv2.ui/save-all-confirmed]}
+  [{:keys [db] :as cofx}]
+  ...)
+```
+
+### Registering top-level re-frame subscriptions
+
+Use `subs.root/reg-root-key-sub` to register top-level (root) subscriptions.
+Additionally, register root subscriptions in the `subs.root` namespace.
+
+```clojure
+;; bad
+(re-frame/reg-sub
+ :view-id
+ (fn [db]
+   (:view-id db)))
+
+;; good
+(reg-root-key-sub :view-id :view-id)
+```
+
+### Registering layer-3 subscriptions
+
+It's extremely important, especially due to performance considerations, to
+define subscriptions that do not directly depend on the entire app db.
+
+The re-frame documentation mentions the [Signal
+Graph](https://day8.github.io/re-frame/subscriptions/#the-four-layers), in
+particular layer-3 subscriptions, also known as _materialized views_ in database
+lingo. We should use them as much as possible.
+
+```clojure
+;; bad
+(re-frame/reg-sub
+ :ens/preferred-name
+ (fn [db]
+   (get-in db [:multiaccount :preferred-name])))
+
+;; good
+(re-frame/reg-sub
+ :ens/preferred-name
+ :<- [:multiaccount]
+ (fn [multiaccount]
+   (:preferred-name multiaccount)))
+```
+
+### Requiring quo2 components
+
+Never directly import a submodule of `quo2`, instead, use the provided
+`quo2.core` namespace.
+
+```clojure
+;; bad
+(ns my-namespace
+  (:require [quo2.components.icon :as icon]))
+
+(icon/icon :i/verified)
+
+;; good
+(ns my-namespace
+  (:require [quo2.core :as quo2]))
+
+(quo2/icon :i/verified)
+```
+
+### Require/import
+
+Prefer `:as` instead of `:refer`. There are exceptions to this rule, e.g. the
+test macros `deftest` and `is`, which are ubiquitous in the Clojure community.
+
+```clojure
+;; bad
+(ns status-im.utils.datetime
+  (:require [cljs-time.coerce :refer [from-long]]))
+
+;; good
+(ns status-im.utils.datetime
+  (:require [cljs-time.coerce :as time.coerce]))
+```
+
+### Accessibility labels
+
+Use keywords instead of strings. As a bonus, remember keywords are cached in
+memory.
+
+```clojure
+;; bad
+[text/text {:accessibility-label "profile-nickname"}
+ "Markov"]
+
+;; good
+[text/text {:accessibility-label :profile-nickname}
+ "Markov"]
+```
+
+### Icons
+
+Use the appropriate keyword qualification/namespace.
+
+```clojure
+(require '[quo2.components.icon :as icons])
+
+;; bad
+(icons/icon :main-icons2/verified)
+
+;; good
+(icons/icon :i/verified)
+```
+
+### Translations
+
+Prefer to use translation placeholders instead of creating multiple translation
+keywords and concatenating them into a single string.
+
+```clojure
+;; bad
+;; Assume the translation key is:
+;;   "biometric-auth-error": "Unable perform biometric authentication"
+(str (i18n/label :t/biometric-auth-error) "(" error-code ")")
+
+;; good
+;; Assume the translation key is:
+;;   "biometric-auth-error": "Unable perform biometric authentication ({{code}})"
+(i18n/label :t/biometric-auth-error {:code error-code})
+```
+
+## Project Structure
```

Comment:
We can add an example for what a single component/screen namespace should look like, as I showed you earlier, where we have the `component_name`, and under it we have 3 files: `view.cljs`, `style.cljs` and `test.cljs`

Reply:
What do you think about this example?

```
src
└── quo2
    ├── components
    │   └── dropdown
    │       ├── style.cljs
    │       ├── test.cljs
    │       └── view.cljs
    └── screens
        └── dropdown
            └── view.cljs
```